### PR TITLE
Structural Refinement and Config Update

### PR DESCRIPTION
**Pull request by ange-black69**
- Using a proper singleton design pattern !
- Prevent player command when joining an arena (you can do /ba, /arena /blockyarena even tho)
- Added a message to broadcast others player when Someone joining a arena channel
- Added a simple confg
- Configurable Lobby Timeout
- Implemented a fully working reload listener 
 -> if you change a node in the config and do sponge plugins reload it update node in memory !

KNOW BUG : 
- No persistant config : If you change a config node and you restart the server, the value is overrided (maybe check if file exists before writting a new empty node ? using loader.load() I dont understand this crap lol )

### CODE REVIEW

File: src/main/java/net/huskycraft/blockyarena/games/Game.java [hunk 1]
```diff
@@ -144,12 +143,17 @@ private void inspect() {
...
-            startingCountdown(10);
+            startingCountdown(BlockyArena.getInstance().getRootNode().getNode("timers", "lobby", "cooldownSec").getInt());
         } else if (timer != null) {
             timer.cancel();
             gameState = GameState.RECRUITING;
             broadcast(Text.of("Waiting for more players to join ..."));
+            
         }
+        else {
+        	MessageChannel.TO_ALL.send((Text)Text.builder("New arena game ! join with /ba join solo !").color(TextColors.GOLD).build());
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/games/Game.java]: I love the idea of broadcasting new available game session to other players. However, we might need to deal with 2v2 so the message needed be adjusted accordingly (`/ba join doubles`).

We might also consider giving players the option to tune out this announcement. Don't send this message to the player that just executed the join command.

File: src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java [hunk 2]
```diff
@@ -37,19 +37,60 @@
...
+		Gamer gamer = GamersManager.getGamer(p.getUniqueId()).get();
+
+		Game game = gamer.getGame();
+		
+		String command = event.getCommand();
+
+		if (game != null)
+		{
+			
+				if (game.getGameState() == GameState.RECRUITING || game.getGameState() == GameState.STARTED || game.getGameState() == GameState.STARTING || game.getGameState() == GameState.STOPPING)
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java:76]: `game.getGameState() != GameState.PLAYING`

Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java]:
```suggestion
				if (gamer.getStatus() == GamerStatus.PLAYING)
```

File: src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java [hunk 3]
```diff
@@ -37,19 +37,60 @@
...
+		
+		String command = event.getCommand();
+
+		if (game != null)
+		{
+			
+				if (game.getGameState() == GameState.RECRUITING || game.getGameState() == GameState.STARTED || game.getGameState() == GameState.STARTING || game.getGameState() == GameState.STOPPING) 
+				{
+						
+					MessageChannel.TO_ALL.send((Text)Text.builder("command issued : " + command).color(TextColors.GOLD).build());
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java]: Do not broadcast commands issued by individual players. May leak sensitive information to others. If this is for testing purposes, remember to mark it as TODO and remove it afterwards.

File: .settings/org.eclipse.buildship.core.prefs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
Comment (mrdarcychen) [inline .settings/org.eclipse.buildship.core.prefs]: add this file to .gitignore

File: .classpath [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mrdarcychen) [inline .classpath]: add this file to .gitignore

File: .project [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mrdarcychen) [inline .project]: add this file to .gitignore

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java [hunk 7]
```diff
@@ -122,15 +148,48 @@ private void createDirectories() {
             }
         }
     }
+    
+    private void HandleConfiguration()
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Consider moving the following to a standalone class (a class dedicated for configuration loading). Right now the main class is overwhelmed.

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java [hunk 8]
```diff
@@ -122,15 +148,48 @@ private void createDirectories() {
...
+    {
+    	
+    	loader = HoconConfigurationLoader.builder().setPath(defaultConfig).build();
+    	
+         this.rootNode = loader.createEmptyNode(ConfigurationOptions.defaults());
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Inconsistent indentation

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java [hunk 9]
```diff
@@ -122,15 +148,48 @@ private void createDirectories() {
...
+    	 rootNode.getNode("timers", "lobby", "cooldownSec").setValue(15);
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Everytime the server inits, this code gets called and thus override the existing value in config to 15. Before setting the default to 15, you may want to check if default config already exists.

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java [hunk 10]
```diff
@@ -109,6 +134,7 @@ private void createManagers() {
     private void createDirectories() {
         arenaDir = Paths.get(getConfigDir().toString() + "/arenas");
         kitDir = Paths.get(getConfigDir().toString() + "/kits");
+        defaultConfig = Paths.get(getConfigDir().toString(), "default.json");
```
Comment (mrdarcychen) [inline src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: To be consistent with Sponge's style, we're using .conf for configs.

**Review by mrdarcychen — CHANGES_REQUESTED** [review]
Thank you for initiating this pull request. Most added features work as expected. Please note the following for your revision in addition to the in-line comments in this pull request:
- Exclude configuration files (dot files) that are only relevant to your local development environment.
- We do not have automated tests to enforce coding style yet, but for now please follow this [guide](https://docs.spongepowered.org/stable/en/contributing/implementation/codestyle.html).]

We'll work together and establish some community guidelines for working on this project later down the road. Cheers!

File: src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java (hunk 2 above)
Comment (ange-black69) [reply src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java]: Fixed !

File: src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java (hunk 3 above)
Comment (ange-black69) [reply src/main/java/net/huskycraft/blockyarena/listeners/EntityListener.java]: Removed !
Was a test until I properly debug !

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java (hunk 7 above)
Comment (ange-black69) [reply src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Done : 
-> ConfigManager Singleton Class
-> create config file if not exist and fill default node value
-> load the conf file if it exist !

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java (hunk 9 above)
Comment (ange-black69) [reply src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Fixed as of new configuration manager !

File: src/main/java/net/huskycraft/blockyarena/BlockyArena.java (hunk 10 above)
Comment (ange-black69) [reply src/main/java/net/huskycraft/blockyarena/BlockyArena.java]: Fixed

**Pull request merged**